app/services/katello/pulp3/repository.rb

```diff
@@ -28,17 +28,17 @@ def with_mirror_adapter
         end
       end
 
-      def self.api(smart_proxy)
+      def self.api(smart_proxy, repo = nil)
```

Comment:
Similarly to my comment above, I think you could override `self.api` (and related methods) in app/services/katello/pulp3/repository/generic.rb so that repository.rb doesn't have to be edited